File: my-folder/1014-k-closest-points-to-origin/solution.py
import heapq

class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        heap = []
        
        for point in points:
            distance = point[0]**2 + point[1]**2
            heapq.heappush(heap, (distance, point))
        
        closest_points = []
        for _ in range(k):
            closest_points.append(heapq.heappop(heap)[1])
        
        return closest_points

File: my-folder/1014-k-closest-points-to-origin/solution.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>(new Comparator<int[]>() {
            @Override
            public int compare(int[] p1, int[] p2) {
                double distance1 = Math.sqrt(p1[0] * p1[0] + p1[1] * p1[1]);
                double distance2 = Math.sqrt(p2[0] * p2[0] + p2[1] * p2[1]);
                return Double.compare(distance1, distance2);
            }
        });
        for (int[] point : points) {
            pq.offer(point);
        }
        int[][] res = new int[k][2];
        for (int i = 0; i < k; i++) {
            res[i] = pq.poll();
        }
        return res;
    }
}
